LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/UpstreamDataFrame.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoRaWan.NetworkServer.BasicsStation
+{
+    public class UpstreamDataFrame
+    {
+        public UpstreamDataFrame(MacHeader mHdr, DevAddr devAddr, FrameControl frameControl, ushort frameCounter, string fOpts, FramePort fPort, string fRMPayload, Mic mic,
+                                 RadioMetadata radioMetadata)
+        {
+            MHdr = mHdr;
+            DevAddr = devAddr;
+            FrameControl = frameControl;
+            FrameCounter = frameCounter;
+            FOpts = fOpts;
+            FPort = fPort;
+            FRMPayload = fRMPayload;
+            Mic = mic;
+            RadioMetadata = radioMetadata;
+        }
+
+        public MacHeader MHdr { get; }
```

Comment:
Suggest using simpler names for the properties (and likewise for the constructor arguments):

```c#
        public MacHeader MacHeader { get; }
        // ...
        public FrameControl Control { get; }
        public ushort Counter { get; }
        public string Options { get; }
        public FramePort Port { get; }
        public string Payload { get; }
```